src/main/groovy/com/blackducksoftware/integration/hub/detect/DetectConfiguration.groovy

```diff
@@ -458,11 +494,47 @@ class DetectConfiguration {
     public String getNoticesReportOutputDirectory() {
         return detectProperties.noticesReportOutputDirectory
     }
+    public String getDockerInspectorAirGapPath() {
+        if (!detectProperties.dockerInspectorAirGapPath?.trim()) {
+            try {
+                File detectJar = new File(guessDetectJarLocation())
+                File inspectorsDirectory = new File(detectJar.getParentFile(), "packaged-inspectors")
+                File dockerAirGapDirectory = new File(inspectorsDirectory, "docker")
+                return dockerAirGapDirectory.getCanonicalPath()
+            } catch (final Exception e) {
+                logger.trace("Exception encountered when guessing air gap path for docker, returning the detect property instead")
+                logger.trace(e.getMessage())
+            }
+        }
+        return detectProperties.dockerInspectorAirGapPath
+    }
     public String getGradleInspectorAirGapPath() {
-        return detectProperties.gradleInspectorAirGapPath?.trim()
+        if (!detectProperties.gradleInspectorAirGapPath?.trim()) {
+            try {
+                File detectJar = new File(guessDetectJarLocation())
+                File inspectorsDirectory = new File(detectJar.getParentFile(), "packaged-inspectors")
+                File gradleAirGapDirectory = new File(inspectorsDirectory, "gradle")
+                return gradleAirGapDirectory.getCanonicalPath()
+            } catch (final Exception e) {
+                logger.trace("Exception encountered when guessing air gap path for gradle, returning the detect property instead")
+                logger.trace(e.getMessage())
+            }
+        }
+        return detectProperties.gradleInspectorAirGapPath
     }
```

Comment:
Looks identical to the other except some constants - can we refactor these to all use the same method?
ie: 
`public String getGradleInspectorAirGapPath() {
    return getInspectorAirGapPath(detectProperties.gradleInspectorAirGapPath,  "gradle");
}`